### ISSUE TITLE
Change package fails for files with a Javadoc comment that includes a method reference with varargs arguments when Maven runs on Java 11+

### ISSUE THREAD

**Issue by vcolin7** [openrewrite/rewrite#3575]
## What version of OpenRewrite are you using?

I am using

- OpenRewrite v5.6.0 (I think? Not sure this version number is the same as the one in the maven plugin.)
- Maven/Gradle plugin v3.8.1
- rewrite-java v8.5.0

## How are you running OpenRewrite?

I am using the Maven plugin on my multi-module project, more specifically the [Azure SDK for Java](https://github.com/Azure/azure-sdk-for-java) as seen in [this PR](https://github.com/Azure/azure-sdk-for-java/pull/36943), but I can also reproduce this issue with a single module project. You can find a sample project with instructions on how to reproduce this [here](https://github.com/vcolin7/openrewrite-varargs-repro).
```xml
<plugin>
  <groupId>org.openrewrite.maven</groupId>
  <artifactId>rewrite-maven-plugin</artifactId>
  <version>5.6.0</version>
  <configuration>
    <activeRecipes>
      <!-- My recipe -->
      <recipe>com.repros.openrewrite</recipe>
    </activeRecipes>
  </configuration>
  <dependencies>
    <dependency>
      <!-- My sample single module project -->
      <groupId>com.repros</groupId>
      <artifactId>openrewrite-varargs</artifactId>
      <version>1.0.0</version>
    </dependency>
  </dependencies>
</plugin>
```

## What is the smallest, simplest way to reproduce the problem?

Having Maven run on Java 11+, try to change the package

```yaml
type: specs.openrewrite.org/v1beta/recipe
name: com.repros.openrewrite
displayName: Create a rewritten class based on an original
recipeList:
  - org.openrewrite.java.ChangePackage:
      oldPackageName: com.original
      newPackageName: com.rewritten
      recursive: true
```

of a class that includes a Javadoc comment that makes a reference to a method that uses varargs in its signature, like the following:
```java
package com.original;

/**
 * This class tests that OpenRewrite can handle varargs in references in Javadoc comments.
 */
public class Original {
    /**
     * A dummy main method. This method is not actually called, but we'll use its Javadoc comment to test that
     * OpenRewrite can handle references like the following: {@link Original#varargsMethod(String...)}.
     *
     * @param args The arguments to the method.
     */
    public static void main(String[] args) {
        System.out.println("Hello, world! This is my original class' main method.");
    }

    /**
     * A method that uses varargs.
     *
     * @param args The arguments to the method.
     */
    public static void varargsMethod(String... args) {
        System.out.println("Hello, world! This is my original class' varargs method.");
    }
}
```
Java 8 works perfectly fine. I saw this failure on Java 11 and Java 17.

## What did you expect to see?

The rewritten class should be in a folder called `rewritten` instead of `original` as shown here:
![image](https://github.com/openrewrite/rewrite/assets/5799331/4456dff4-d1f2-4366-8dd9-f84f2cb9648f)

## What did you see instead?

The rewritten class was in a folder with the original name: `original` as shown here:
![image](https://github.com/openrewrite/rewrite/assets/5799331/b7497df3-890e-4ae1-b4f3-c0830e4c33e2)
> Note: the screenshot shows two folders because for testing I use a PowerShell script that copies my original folder to avoid rewriting on top of it.

## What is the full stack trace of any errors you encountered?

```
[INFO] --- rewrite-maven-plugin:5.6.0:run (default-cli) @ my-rewritten-project ---
[INFO] Using active recipe(s) [com.repros.openrewrite]
[INFO] Using active styles(s) []
[INFO] Validating active recipes...
[INFO] Project [my-rewritten-project] Resolving Poms...
[ERROR] Project [my-rewritten-project] Parse resulted in no Maven source files. Maven Project File 'C:\Users\mogli\Documents\GitHub\openrewrite-varargs-repro\openrewrite\..\my-rewritten-project\pom.xml'
[INFO] Project [my-rewritten-project] Parsing source files
[WARNING] There were problems parsing some source files, run with --errors to see full stack traces
[WARNING] There were problems parsing my-rewritten-project\src\com\original\Original.java
[INFO] Running recipe(s)...
[WARNING] File has been moved from my-rewritten-project\src\com\original\Original.java to my-rewritten-project\src\com\original\Rewritten.java by:
[WARNING]     com.repros.openrewrite
[WARNING]         org.openrewrite.RenameFile: {fileMatcher=**/Original.java, fileName=Rewritten.java}
[WARNING]         org.openrewrite.text.FindAndReplace: {find=com.original, replace=com.rewritten, caseSensitive=true, filePattern=**/*.java;**/*.xml;**/*.md;**/*.properties;**/*.json}
[WARNING]         org.openrewrite.text.FindAndReplace: {find=original, replace=rewritten, caseSensitive=true, filePattern=**/*.java;**/*.xml;**/*.md}
[WARNING]         org.openrewrite.text.FindAndReplace: {find=Original, replace=Rewritten, caseSensitive=true, filePattern=**/*.java;**/*.xml;**/*.md}
[WARNING] File has been moved from my-rewritten-project\my-original-project.iml to my-rewritten-project\my-rewritten-project.iml by:
[WARNING]     com.repros.openrewrite
[WARNING]         org.openrewrite.RenameFile: {fileMatcher=**/my-original-project.iml, fileName=my-rewritten-project.iml}
[WARNING] Please review and commit the results.
```

When things go right (not using a Javadoc comment referring to a method that accepts varargs) it looks like this:
```
[INFO] --- rewrite-maven-plugin:5.6.0:run (default-cli) @ my-rewritten-project ---
[INFO] Using active recipe(s) [com.repros.openrewrite]
[INFO] Using active styles(s) []
[INFO] Validating active recipes...
[INFO] Project [my-rewritten-project] Resolving Poms...
[ERROR] Project [my-rewritten-project] Parse resulted in no Maven source files. Maven Project File 'C:\Users\mogli\Documents\GitHub\openrewrite-varargs-repro\openrewrite\..\my-rewritten-project\pom.xml'
[INFO] Project [my-rewritten-project] Parsing source files
[INFO] Running recipe(s)...
[WARNING] File has been moved from my-rewritten-project\src\com\original\Original.java to my-rewritten-project\src\com\rewritten\Rewritten.java by:
[WARNING]     com.repros.openrewrite
[WARNING]         org.openrewrite.java.ChangePackage: {oldPackageName=com.original, newPackageName=com.rewritten, recursive=true}
[WARNING]         org.openrewrite.RenameFile: {fileMatcher=**/Original.java, fileName=Rewritten.java}
[WARNING]         org.openrewrite.text.FindAndReplace: {find=original, replace=rewritten, caseSensitive=true, filePattern=**/*.java;**/*.xml;**/*.md}
[WARNING]         org.openrewrite.text.FindAndReplace: {find=Original, replace=Rewritten, caseSensitive=true, filePattern=**/*.java;**/*.xml;**/*.md}
[WARNING] File has been moved from my-rewritten-project\my-original-project.iml to my-rewritten-project\my-rewritten-project.iml by:
[WARNING]     com.repros.openrewrite
[WARNING]         org.openrewrite.RenameFile: {fileMatcher=**/my-original-project.iml, fileName=my-rewritten-project.iml}
[WARNING] Please review and commit the results.
```

## Are you interested in [contributing a fix to OpenRewrite](https://github.com/openrewrite/.github/blob/main/CONTRIBUTING.md#contributing-fixes)?

I would love to but I'm not sure I have the bandwidth at the moment.

**Comment by timtebeek** [comment]
hi @vcolin7 ! Neat seeing you here and looking to apply changes broadly to the Azure SDK. We'd love to help you get going!

Thanks for the detailed report and reproduction sample. I've been able to distill it down to a problem with our JavaDoc parser, by adding the following test to https://github.com/openrewrite/rewrite/blob/a60b83a92c789ee838a0caed02ca74774141930b/rewrite-java-tck/src/main/java/org/openrewrite/java/tree/JavadocTest.java#L25

```java
@Test
@Issue("https://github.com/openrewrite/rewrite/issues/3575")
void varargsMethod() {
    rewriteRun(
      java(
        """
          class A {
              /**
               * A dummy main method. This method is not actually called, but we'll use its Javadoc comment to test that
               * OpenRewrite can handle references like the following: {@link A#varargsMethod(String...)}.
               *
               * @param args The arguments to the method.
               */
              public static void main(String[] args) {
                  System.out.println("Hello, world! This is my original class' main method.");
              }
              public static void varargsMethod(String... args) {
                  System.out.println("Hello, world! This is my original class' varargs method.");
              }
          }
          """
      )
    );
}
```

You can see that reflected in your log output above around:
```
[WARNING] There were problems parsing some source files, run with --errors to see full stack traces
[WARNING] There were problems parsing my-rewritten-project\src\com\original\Original.java
```
In short that says: we failed to parse that file, and therefor we can't make any changes to it.

We'll need to fix the parsing issue, and having the above test should already help.

**Comment by timtebeek** [comment]
Likely fixed in
- https://github.com/openrewrite/rewrite/pull/3577

Thanks for bringing this to our attention! Really helps to have a detailed report.

**Comment by timtebeek** [comment]
Fixed and released in https://github.com/openrewrite/rewrite/releases/tag/v8.6.2
Hope that helps!

**Comment by vcolin7** [comment]
Hey @timtebeek, I checked things again and saw that this was only happening on Java 17 and not on Java 11 as I had mistakenly pointed out. Even though this was marked as resolved, I updated the title and description just in case someone comes across this in the future.

**Comment by timtebeek** [comment]
Yes it turns out only our `ReloadableJava17JavadocVisitor.JavaVisitor` has a `visitArrayType` method; the implementations for 8 and 11 fully miss that method, and therefor did not throw any such exception. Not quite sure if we should back port that method to those implementations, but good to know that at least on 17 we don't have that issue anymore.
